lib/electrode-server.js

```diff
@@ -0,0 +1,347 @@
+        }
+        try {
+          mod = xrequire(name);
+        } catch (error) {
+          error.message = `Failed loading module ${pluginMod.name}${fromPath}: ${error.message}`;
+          throw error;
+        }
+      };
+
+      requireMod();
+
+      // order of fields to look for the Fastify plugin from the module:
+      // 1. mod.register
+      // 4. mod.default (ES6)
+      // 5. mod
+      const plugin = ["plugin", "default"].find(x => _.get(mod, x));
+      // validate plugin
+      p.register = (plugin && _.get(mod, plugin)) || mod;
+
+      const validatePlugin = pos => {
+        assert(pos.register, `Plugin ${name} is a falsy value: ${pos.register}`);
+        assert(typeof pos.register === "function", `Plugin ${name} should be a function`);
+      };
+      validatePlugin(p);
+      return p;
+    };
+
+    return doRequire();
+  };
+
+  const num = x => {
+    return _.isString(x) ? parseInt(x, 10) : x;
+  };
+
+  const checkNaN = x => {
+    return isNaN(x) ? Infinity : x;
+  };
+
+  const priority = p => checkNaN(num(p.priority));
+  const isEnable = p => p.__name !== "requireFromPath" && p.enable !== false;
+  const transpose = (p, k) => Object.assign({ __name: k }, p);
+
+  //
+  // transpose each plugin, filter out disabled ones, and sort by priority
+  //
+  const pluginsArray = () =>
+    _(plugins)
+      .map(transpose)
+      .filter(isEnable)
+      .sortBy(priority)
+      .value();
+  // convert plugins object to array and check each one if it has a module to load.
+  const arr = pluginsArray();
+  return xaa.map(arr, loadModule);
+}
+
+async function startElectrodeServer(context) {
+  const server = context.server;
+  const config = context.config;
+  let started = false;
+
+  const registerPlugins = async plugins => {
+    const errorRegisterMessage = plg => {
+      if (plg.module) {
+        const fromPath = plg.requireFromPath ? ` from path: '${plg.requireFromPath}'` : "";
+        return `with module '${JSON.stringify(plg.module)}'${fromPath}`;
+      } else {
+        return `with register function`;
+      }
+    };
+    const registerPromises = [];
+    plugins.forEach(plg => {
+      server.register(plg.register, plg.options);
+      const registerStatus = new Promise((resolve, reject) => {
+        server.after(err => {
+          if (err) {
+            if (!err.hasOwnProperty("message")) {
+              err = new Error(err);
+            } else if (err.code === "ERR_AVVIO_PLUGIN_TIMEOUT") {
+              err.message = `plugin '${
+                plg.__name
+              }' with register function timeout - did you return a resolved promise?`;
+            }
+
+            err.code = "XPLUGIN_FAILED";
+            err.plugin = plg;
+            err.method = errorRegisterMessage(plg);
+
+            reject(err);
+          }
+          resolve();
+        });
+      });
+      registerPromises.push(registerStatus);
+    });
+    await server.ready();
+    await Promise.all(registerPromises);
```

Comment:
Are you sure this is needed? `ready` callback is called only after all plugins are registered.

Reply:
True.  Wanted to capture the plugin exceptions and display a nice message.  Otherwise they go as unhandled exceptions.